### PR TITLE
BOTMETA: enforce entries for new plugins/modules, add documentation for creating new plugins/modules

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
This is a follow-up to #3069: fixes some details I forgot, adds documentation on how to add new plugins/modules which mentions BOTMETA, and enables sanity test which enforces this.

##### ISSUE TYPE
- Docs Pull Request
- CI Pull Request

##### COMPONENT NAME
BOTMETA
BOTMETA sanity test

### CODE REVIEW

**Comment by felixfontein** [comment]
CC @Andersson007

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -34,3 +34,34 @@ You can also read [our Quick-start development guide](https://github.com/ansible
...
+   - Follow [documentation standards](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html) and
+     the [Ansible style guide](https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide);
+   - Make sure your modules and plugins are [GPL-3.0-or-later](https://www.gnu.org/licenses/gpl-3.0-standalone.html) licensed
+     (new module_utils can also be [BSD-2-clause](https://opensource.org/licenses/BSD-2-Clause) licensed);
+   - Make sure that new plugins and modules have tests (unit tests, integration tests, or both); it is preferable to have some tests
+     which run in CI.
+
+3. For modules and action plugins, make sure to create your module/plugin in the correct subdirectory, and create a symbolic link
+   from `plugins/modules/` respectively `plugins/action/` to the actual module/plugin code. (Other plugin types should not use
+   subdirectories.)
```
Comment (Andersson007) [inline CONTRIBUTING.md:57]:
```suggestion
   from `plugins/modules/` respectively `plugins/action/` to the actual module/plugin code (other plugin types should not use
   subdirectories).
```

Comment (Andersson007) [reply CONTRIBUTING.md:57]: Just a suggestion, feel free to ignore if you think the original variant is better

**Review by Andersson007 — APPROVED** [review]
Thanks! I wonder if we should copy the changes of CONTRIBUTING.md to community-docs, I like it, spots important things

File: CONTRIBUTING.md (hunk 1 above)
Comment (felixfontein) [reply CONTRIBUTING.md:57]: I think it's easier to read as a separate sentence. The first sentence is already pretty long anyway (and probably too long :) ).

Comment (Andersson007) [reply CONTRIBUTING.md:57]: sounds sensible:) I'm just not 100% sure if the parentheses are allowed there. Though never mind:)

**Review by Andersson007 — COMMENTED** [review]
LGTM

**Comment by felixfontein** [comment]
@Andersson007 thanks for reviewing this!

**Pull request merged**